### PR TITLE
Remove deprecated getAntBuilder method

### LINKED ISSUE #22367: Remove deprecated getAntBuilder from Checkstyle/Pmd/CodeNarc

### PR DESCRIPTION
**Pull request by tresat**
Fixes #22367

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/architecture-test/src/changes/accepted-public-api-changes.json [hunk 1]
```diff
@@ -296,6 +296,14 @@
                 "Method has been removed"
             ]
         },
+        {
+            "type": "org.gradle.api.plugins.quality.Checkstyle",
+            "member": "Method org.gradle.api.plugins.quality.Checkstyle.getAntBuilder()",
+            "acceptation": "[ADD YOUR CUSTOM REASON HERE]",
```
Comment (big-guy) [inline subprojects/architecture-test/src/changes/accepted-public-api-changes.json:302]: Please add a reason.

**Review by big-guy — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**